src/call/mod.rs

```diff
@@ -766,6 +841,63 @@ impl SinkBase {
 mod tests {
     use super::*;
 
+    #[test]
+    fn byte_buffer_empty() {
+        let mut buf = GrpcByteBuffer::default();
+        unsafe {
+            GrpcByteBuffer {
+                raw: buf.take_raw()
+            }.len();
+        }
+        buf.len();
+    }
+
+    #[test]
+    fn byte_buffer_clear_after_taken_away() {
+        let mut buf = GrpcByteBuffer::default();
+        buf.push(&mut From::from("oh my god!".as_bytes()));
+        unsafe {
+            GrpcByteBuffer {
+                raw: buf.take_raw()
+            }.len();
+        }
+        buf.clear();
+        buf.len();
+    }
+
+    #[test]
+    fn byte_buffer_clear_empty() {
+        let mut buf = GrpcByteBuffer::default();
+        buf.clear();
+        buf.clear();
```

Comment:
To see if `clear` works when there's nothing inside a bytebuf. `clear` is reset and unref, I just want to ensure this behavior persists in the future because I depend on this logic